internal/exec/atlantis_generate_repo_config.go

```diff
@@ -0,0 +1,236 @@
+func ExecuteAtlantisGenerateRepoConfig(configTemplateName string, projectTemplateName string, workflowTemplateName string, stacks []string, components []string) error {
+	var configAndStacksInfo c.ConfigAndStacksInfo
+	stacksMap, err := FindStacksMap(configAndStacksInfo, false)
+	if err != nil {
+		return err
+	}
+
+	var configTemplate c.AtlantisRepoConfig
+	var projectTemplate c.AtlantisProjectConfig
+	var workflowTemplate any
+	var ok bool
+
+	if configTemplate, ok = c.Config.Integrations.Atlantis.ConfigTemplates[configTemplateName]; !ok {
+		return errors.Errorf("atlantis config template '%s' is not defined in 'integrations.atlantis.config_templates' in atmos.yaml", configTemplateName)
+	}
+
+	if projectTemplate, ok = c.Config.Integrations.Atlantis.ProjectTemplates[projectTemplateName]; !ok {
+		return errors.Errorf("atlantis project template '%s' is not defined in 'integrations.atlantis.project_templates' in atmos.yaml", projectTemplateName)
+	}
+
+	if workflowTemplate, ok = c.Config.Integrations.Atlantis.WorkflowTemplates[workflowTemplateName]; !ok {
+		return errors.Errorf("atlantis workflow template '%s' is not defined in 'integrations.atlantis.workflow_templates' in atmos.yaml", workflowTemplateName)
+	}
+
+	var atlantisProjects []c.AtlantisProjectConfig
+	atlantisWorkflows := map[string]any{}
+
+	// Iterate over all components in all stacks and generate atlantis projects
+	for stackConfigFileName, stackSection := range stacksMap {
+		if componentsSection, ok := stackSection.(map[any]any)["components"].(map[string]any); ok {
+			if terraformSection, ok := componentsSection["terraform"].(map[string]any); ok {
+				for componentName, compSection := range terraformSection {
+					if componentSection, ok := compSection.(map[string]any); ok {
+						// Find all derived components of the provided components
+						derivedComponents, err := s.FindComponentsDerivedFromBaseComponents(stackConfigFileName, terraformSection, components)
+						if err != nil {
+							return err
+						}
+
+						if len(components) == 0 || u.SliceContainsString(components, componentName) || u.SliceContainsString(derivedComponents, componentName) {
+							if varsSection, ok := componentSection["vars"].(map[any]any); ok {
+								// Component metadata
+								metadataSection := map[any]any{}
+								if metadataSection, ok = componentSection["metadata"].(map[any]any); ok {
+									if componentType, ok := metadataSection["type"].(string); ok {
+										// Don't include abstract components
+										if componentType == "abstract" {
+											continue
+										}
+									}
+								}
+
+								// Find terraform component
+								// If `component` attribute is present, it's the terraform component
+								// Otherwise, the YAML component name is the terraform component
+								terraformComponent := componentName
+								if componentAttribute, ok := componentSection["component"].(string); ok {
+									terraformComponent = componentAttribute
+								}
```

Comment:
This logic is used in other places. It would be good to set the `component` key to the YAML component name if the `component` key is not already set explicitly when constructing the `componentSection` map